### ISSUE TITLE
Image fields' URLs aren't being correctly generated (prepended with 6 file://<home directory path>)

### ISSUE THREAD

**Issue by arturmatos** [incuna/django-wkhtmltopdf#22]
I have a Django model with a ImageField, and I'm trying to use PDFTemplateView to generate a PDF that includes that image. However, the image URL isn't being properly generated in the PDF. If I put  something like this in my template:

<p>{{ user.get_profile.company.logo.url }}</p>

This gets properly converted if I request it in HTML (with ?as=html):

<p>/media/logos/abcfinancialservices.png</p>

But if I'm generating the PDF I get this instead:

file:///Users/amatosfile:///Users/amatosfile:///Users/amatosfile:///Users/amatosfile:///Users/amatos/media/logos/abcfinancialservices.png

So basically I'm getting my home directory prepended 6 times in the URL.

I haven't had more time to look into it, but it looks like a bug. Or I'm doing anything wrong?

I'm not that familiar with Django, but assuming it's a bug I'm happy to submit a patch if I can get some guidance -I just need some brief explanation of what the problem might be.

**Comment by tachang** [comment]
Can someone explain the regex pattern:

        occur_pattern = '''["|']({0}.*?)["|']'''

I am a bit confused. It seems to be picking up more than it should.

**Comment by tachang** [comment]
My MEDIA_ROOT and MEDIA_URL were blank so it caused extra stuff to be picked up (because of the dynamic regex). I fixed it in my fork.

**Comment by metaljoe** [comment]
The project I've started working on doesn't use/need MEDIA_URL or MEDIA_ROOT set, so I've had a similar problem caused by that regex. Because they're still on the defaults (empty strings), the content had a lot of "file:///"s inserted everywhere. As a workaround, I've set MEDIA_URL for the moment - tachang's fix looks like it does the job.

**Comment by fonfon** [comment]
I ran into the `MEDIA_URL` problems as well.

Additionally, I'm generating images on-the-fly on the server side, so `make_absolute_paths` is not useful for me because it always puts in the `file://` prefix.

My solution is to monkey-patch `make_absolute_paths` with a custom function. There I replace the path of my server-side generated images by the full URL. This works fine, but it's still monkey patching..

It would be cool to be able to customize the `overrides` in the settings -- to an extent that allows using both `file:` prefixes as well as `http://` or whatever one would like. The current overrides could still be the default ones.

**Comment by johnraz** [comment]
@tachang do you mind submitting a PR for this ?

**Comment by johnraz** [comment]
@tachang another chance?

**Comment by amirkdv** [comment]
I just got bitten by the same problem. Debugging made clear that the real issue is the way the regular expression in `make_absolute_paths` is capturing matching patterns.

Example broken test:
```py
# test/__init__.py
```
```py
# test_22/settings.py
SECRET_KEY = 'x' # make django not crash

STATIC_ROOT = '/path/to/some/dir'
STATIC_URL = '/static/'
```
```py
# test_22/test.py
from wkhtmltopdf.utils import make_absolute_paths

content = """
    <img src="/static/foo.png"/>
    <img src="/static/bar/static/foo.png"/>
"""
print(make_absolute_paths(content))
```
Result (depending on the order or matches returned by `re` either of the following):
```sh
$ DJANGO_SETTINGS_MODULE=test_22.settings python manage.py shell < test_22/test.py

    <img src="file:///path/to/some/dir/foo.png"/>
    <img src="/static/barfile:///path/to/some/dir/foo.png"/>

$ DJANGO_SETTINGS_MODULE=test_22.settings python manage.py shell < test_22/test.py

    <img src="file:///path/to/some/dir/foo.png"/>
    <img src="file:///path/to/some/dir/barfile:///path/to/some/dir/foo.png"/>
```
Note the consistently broken second URL.

What happens is this:
* The regex matches `"/static/foo.png"` but only captures `/static/foo.png` (does not capture quotes).
* Then when string replace is called all instances of the captured string are replaced which includes parts of the second URL.
* This can be fixed by capturing the quotes in the regular expression as well.

I have sent PR https://github.com/incuna/django-wkhtmltopdf/pull/167 that solves this by capturing the quotes and then putting them back in string replace.

Results:
```sh
$ DJANGO_SETTINGS_MODULE=test_22.settings python manage.py shell < test_22/test.py

    <img src="file:///path/to/some/dir/foo.png"/>
    <img src="file:///path/to/some/dir/bar/static/foo.png"/>
```
